crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -136,29 +136,63 @@ impl Solution {
         let mut trades = Vec::with_capacity(self.trades.len());
-        for trade in self.user_trades() {
-            // Solver generated fulfillment does not include the fee in the executed amount
-            // for sell orders.
-            let executed = match trade.order().side {
-                order::Side::Sell => (trade.executed().0 + trade.fee().0).into(),
-                order::Side::Buy => trade.executed(),
-            };
-            let uniform_prices = ClearingPrices {
-                sell: self
-                    .clearing_price(trade.order().sell.token)
-                    .ok_or(error::Scoring::InvalidClearingPrices)?,
-                buy: self
-                    .clearing_price(trade.order().buy.token)
-                    .ok_or(error::Scoring::InvalidClearingPrices)?,
-            };
-            trades.push(scoring::Trade::new(
-                trade.order().sell,
-                trade.order().buy,
-                trade.order().side,
-                executed,
-                trade.custom_prices(&uniform_prices)?,
-                trade.order().protocol_fees.clone(),
-            ))
+        for trade in self.trades() {
+            match trade {
+                Trade::Fulfillment(fulfillment) => {
+                    // Solver generated fulfillment does not include the fee in the executed amount
+                    // for sell orders.
+                    let executed = match fulfillment.order().side {
+                        order::Side::Sell => {
+                            (fulfillment.executed().0 + fulfillment.fee().0).into()
+                        }
+                        order::Side::Buy => fulfillment.executed(),
+                    };
+                    let uniform_prices = ClearingPrices {
+                        sell: self
+                            .clearing_price(fulfillment.order().sell.token)
+                            .ok_or(error::Scoring::InvalidClearingPrices)?,
+                        buy: self
+                            .clearing_price(fulfillment.order().buy.token)
+                            .ok_or(error::Scoring::InvalidClearingPrices)?,
+                    };
+                    trades.push(scoring::Trade::new(
+                        fulfillment.order().sell,
+                        fulfillment.order().buy,
+                        fulfillment.order().side,
+                        executed,
+                        fulfillment.custom_prices(&uniform_prices)?,
+                        fulfillment.order().protocol_fees.clone(),
+                    ))
+                }
+                Trade::Jit(jit) => {
+                    if self
+                        .solver
+                        .cow_amm_addresses()
+                        .contains(&jit.order().signature.signer.into())
+                    {
+                        let executed = match jit.order().side {
```

Comment:
We can potentially move the functions at `Trade` level and do this operation at such level.